Backend

OliverNeuhaeusler opened the pull request:
Hello. 
please  test my search favorite, comment and Ratingfunction. i put in the backend so please look if it works.
Thank your for reviewing

OliverNeuhaeusler committed:
put in the search function

OliverNeuhaeusler committed:
put in the backendlogic

OliverNeuhaeusler committed:
create search function and fix commit and rating function

OliverNeuhaeusler committed:
put in the the logic for backend update markets

OliverNeuhaeusler committed:
some small changes

OliverNeuhaeusler committed:
put in the Rating and Comment put logic.

miriamhaenle commented on client/package.json:
```diff
@@ -35,5 +35,6 @@
     ]
-  }
+  },
+  "porxy": "http://localhost:4000"
```
You probably meant 
```suggestion
  "proxy": "http://localhost:4000"
```

miriamhaenle commented on client/src/App.js:
```diff
@@ -3,31 +3,143 @@ import { Switch, Route } from 'react-router-dom';
 import { saveToLocalStorage, loadFromLocalStorage } from './lib/localStorage';
 import styled from 'styled-components/macro';
 import Home from './pages/Home.js';
-import BurgerMenu from './BurgerMenu.js';
+import BurgerMenu from './components/BurgerMenu.js';
 import MarketCard from './pages/MarktCard.js';
 import MarketForm from './pages/MarktForm.js';
```
It is often seen as good practice to structure the imports from outside in:
First external imports (e.g. `styled`)
And then your own components, libs etc. I would also recommend that you group them (all pages together) and sort them alphabetically. Just gives a very structured impression :)

OliverNeuhaeusler replied:
Thnak you Miriam i changed it now

miriamhaenle commented on client/src/App.js:
```diff
@@ -3,31 +3,143 @@ import { Switch, Route } from 'react-router-dom';
+  }
 
-    setMarkets(updatedMarkets);
+  function addRating(rating, marketToUpdate) {
```
Your functions `addComment` and `addRating` are very similar. The only distinction being, that the one updates the ratings and the other the comments, but the rest of the code is almost identical. 
Keeping the DRY principles in mind, this calls for refactoring 😄 Maybe you find a way to improve this to have one function that handles both use cases. Not something that needs to be done in this PR but keep that as possible improvement for a future sprint.

miriamhaenle commented on client/src/App.js:
```diff
@@ -3,31 +3,143 @@ import { Switch, Route } from 'react-router-dom';
+  }, []);
+
+  useEffect(() => {
+    setFilteredMarkets(markets);
```
Why the additional `useEffect` here?
Both your state update function (`setFilteredMarkets`) and your local storage function (`saveToLocalStorage`) have the `markets` as dependency array. So you could put both into one `useEffect` :)

miriamhaenle commented on client/src/App.js:
```diff
@@ -3,31 +3,143 @@ import { Switch, Route } from 'react-router-dom';
     saveToLocalStorage('Markets', markets);
   }, [markets]);
 
+  useEffect(() => {
```
Your structure is a bit of a mixture of useEffects and functions. 
I would recommend that you put all hooks on top of your component (`useState` and `useEffect`) .
Afterwards the functions can follow. 
But also consider extracting the functions from the file and importing them :)

miriamhaenle commented on client/src/App.js:
```diff
@@ -42,11 +154,32 @@ function App() {
           <Route path="/market">
-            {markets.map((market) => (
-              <MarketCard market={market} onAddComment={addComment} />
+            <SearchboxInput
+              type="text"
+              placeholder="Suche hier deinen Markt."
+              onChange={searchedMarkets}
+            />
+            {filteredMarkets &&
+              filteredMarkets.map((filteredMarket) => (
+                <MarketCard
+                  market={filteredMarket}
+                  onAddComment={addComment}
+                  onAddRating={addRating}
+                  onAddToFav={toggleFav}
+                  isFavorite={isFavorite}
+                />
+              ))}
```
Could be extracted to a page to keep the `App.js` smaller and hide the details within that component/page

miriamhaenle commented on client/src/App.js:
```diff
@@ -42,11 +154,32 @@ function App() {
+              ))}
+          </Route>
+          <Route path="/favorites">
+            {bookmarkedMarkets.map((market) => (
+              <MarketCard
+                market={market}
+                onAddComment={addComment}
+                onAddToFav={toggleFav}
+                isFavorite={isFavorite}
```
Same as previous comment :)

miriamhaenle commented on client/src/components/Login.js:
```diff
@@ -0,0 +1,50 @@
+}
+
+const LoginForm = styled.section`
+  display: fles;
```
Haha, should not actually be a flexbox at the moment :P 
```suggestion
  display: flex;
```

miriamhaenle commented on client/src/components/Login.js:
```diff
@@ -0,0 +1,50 @@
+  return (
+    <LoginForm>
+      <form onSubmit={handleSubmit}>
+        <article size="lg" controlId="email">
```
What do you use the `size` attribute for?

miriamhaenle reviewed (changes requested):
Hi Oli, really great job you did here 👏  

I only had a few remarks. Your App.js is pretty huge at the moment, see what you maybe can extract from there to make it a bit more concise.

Keep up the good work :)

OliverNeuhaeusler committed:
Update client/package.json

Co-authored-by: Miriam Hänle <60315975+miriamhaenle@users.noreply.github.com>

OliverNeuhaeusler committed:
Update client/src/components/Login.js

i changed it but this login should not be in this branch. so i delete it an put it in the rcorrect branch

Co-authored-by: Miriam Hänle <60315975+miriamhaenle@users.noreply.github.com>

OliverNeuhaeusler committed:
some small refactoring

OliverNeuhaeusler committed:
put back in the favlogic

OliverNeuhaeusler committed:
put the Search and Fav in components

OliverNeuhaeusler commented:
Thank you for reviewing my Code.  iam glad that you like it and i changed the things you said.

OliverNeuhaeusler committed:
refactored addComment and addRating

anitaschiller reviewed (approved):
Good work and nice refactoring, can be merged! :-)
